### PR TITLE
Bug 1839640 - BigQuery: Convert json atoms to a JSON column type

### PR DESCRIPTION
**Pull request by badboy**
I _think_ this is all we need here? At least it works in my hacky schema locally.

### CODE REVIEW

**Review by fbertsch — COMMENTED** [review]
Backwards compatibility will be the real pain here, although we may not actually be generating any JSON fields at the moment. One way to do check is to run `mozilla-schema-generator` with your changes here and compare the generated schemas to our prod runs, and verify that they match.

The docs could also use updated. It's not clear to me from this change when a JSON column gets reported, since technically everything being passed in is JSON.

**Comment by badboy** [comment]
> Backwards compatibility will be the real pain here, although we may not actually be generating any JSON fields at the moment. One way to do check is to run `mozilla-schema-generator` with your changes here and compare the generated schemas to our prod runs, and verify that they match.
> 
> The docs could also use updated. It's not clear to me from this change when a JSON column gets reported, since technically everything being passed in is JSON.

yeah, as can be seen from the test failure there's definitely some edge cases and it's unclear whether those can occur in real code from us. in the worst case we might need to restrict this a bit further.

**Comment by badboy** [comment]
After talking with akomar we came to the conclusion that we need some metadata configuration to pass down which fields get to be JSON columns.
The idea is to put that into `mozPipelineMetadata`, which this PR now implements. I'm still testing this out across the pipeline tools.

File: src/main.rs [hunk 1]
```diff
@@ -81,6 +81,7 @@ fn main() {
         force_nullable: args.force_nullable,
         tuple_struct: args.tuple_struct,
         allow_maps_without_value: args.allow_maps_without_value,
+        json_object_path: None,
```
Comment (akkomar) [inline src/main.rs:84]: This should be passed as a parameter.
How about calling it `json_object_path_regex`?

File: build.rs [hunk 2]
```diff
@@ -86,7 +88,12 @@ fn avro_{name}() {{
     }};
     let input: Value = serde_json::from_str(input_data).unwrap();
     let expected: Value = serde_json::from_str(expected_data).unwrap();
-    assert_eq!(expected, convert_avro(&input, context));
+    if expected.is_null() {{
+        // No expected data = no avro support
```
Comment (akkomar) [inline build.rs:92]: How about we throw an exception or at least a warning when Avro transpiler is run with object metric path?
Although I filed a [ticket](https://mozilla-hub.atlassian.net/browse/DENG-2029) to get rid of Avro support, I don't know when this happens.

**Review by akkomar — COMMENTED** [review]
Added comments I already sent on Slack.

File: src/main.rs (hunk 1 above)
Comment (badboy) [reply src/main.rs:84]: As a parameter to `jsonschema-transpiler`? 
Then we will have to configure it differently in MSG.
With this change as is we parse this config from the schema itself, so every invocation of jsonschema-transpiler, no matter where, will just pick it up (and apparently there's at least a couple of other invocations of jst, e.g. in MPS for testing)

Comment (akkomar) [reply src/main.rs:84]: > we parse this config from the schema itself, so every invocation of jsonschema-transpiler, no matter where, will just pick it up

I see, I haven't realized that, this makes sense then!

File: build.rs (hunk 2 above)
Comment (badboy) [reply build.rs:92]: Oh, I guess I can add another test case that catches this.
Because `avro` will already panic, because it doesn't know what to do with the JSON piece in the schema

**Review by akkomar — APPROVED** [review]
